backend/src/pilstark/json_exporter/mod.rs

```diff
@@ -385,12 +393,54 @@ mod test {
             panic!("Pilcom did not generate {output_file:?} at the expected location.")
         });
         drop(temp_dir);
-        let pilcom_parsed = json::parse(&pilcom_out).expect("Invalid json from pilcom.");
+
+        let json_out = serde_json::to_value(pil_out).unwrap();
+
+        let mut pilcom_parsed =
+            serde_json::from_str(&pilcom_out).expect("Invalid json from pilcom.");
+
+        // Filter out expression's "deps" before comparison, since we don't
+        // export them.
+        filter_out_deps(&mut pilcom_parsed);
+
         (json_out, pilcom_parsed)
     }
 
+    fn filter_out_deps(value: &mut serde_json::Value) {
+        match value {
+            JsonValue::Array(arr) => {
+                for e in arr {
+                    filter_out_deps(e);
+                }
+            }
+            JsonValue::Object(obj) => {
+                match obj.entry("deps") {
+                    serde_json::map::Entry::Occupied(deps) => {
+                        deps.remove();
+                    }
+                    _ => (),
+                }
+
+                for (_, e) in obj.iter_mut() {
+                    filter_out_deps(e);
+                }
+            }
+            _ => (),
+        }
+    }
+
     fn compare_export_file(file: &str) {
         let (json_out, pilcom_parsed) = generate_json_pair(file);
+        serde_json::to_writer_pretty(
+            std::io::BufWriter::new(fs::File::create("/tmp/ours.json").unwrap()),
```

Comment:
This should use `tempdir`

Reply:
This is debug leftover that should never have been committed.

Reply:
what has been seen cannot be unseen